clients/gtest/gemm_ex_gtest.cpp

```diff
@@ -146,6 +146,203 @@ const vector<vector<int>> NaN_matrix_size_range = {
+    {  256,  1024,   196,   196,   196,   256,   256},
+    {  256,  1024,    49,    49,    49,   256,   256},
+    {  256,   128,   784,   784,   784,   256,   256},
+    {  256,   256,   196,   196,   196,   256,   256},
+    {  256,    64,  3136,  3136,  3136,   256,   256},
+    {   25,    64, 27920, 27920, 27920,    25,    25},
+    {   27,    64,  2916,  2916,  2916,    27,    27},
+    {   27,    64, 50176, 50176, 50176,    27,    27},
+    {  288,    64,  1440,  1440,  1440,   288,   288},
+    { 3200,   256,  1653,  1653,  1653,  3200,  3200},
+    { 4608,   512,   196,   196,   196,  4608,  4608},
+    { 4608,   512,   420,   420,   420,  4608,  4608},
+    { 4608,   512,    49,    49,    49,  4608,  4608},
+    { 4800,    32,   784,   784,   784,  4800,  4800},
+    {  512,  1024,   196,   196,   196,   512,   512},
+    {  512,   128,   784,   784,   784,   512,   512},
+    {  512,   192,   196,   196,   196,   512,   512},
+    {  512,  2048,    49,    49,    49,   512,   512},
+    {  512,   256,   196,   196,   196,   512,   512},
+    {  512,   512,    49,    49,    49,   512,   512},
+    {  576,   128, 12544, 12544, 12544,   576,   576},
+    {  576,   128,   360,   360,   360,   576,   576},
+    {  576,    64, 28000, 28000, 28000,   576,   576},
+    {  576,    64,  2916,  2916,  2916,   576,   576},
+    {  576,    64,  3136,  3136,  3136,   576,   576},
+    { 6400,   512,   369,   369,   369,  6400,  6400},
+    {   64,   256,  3136,  3136,  3136,    64,    64},
+    {   64,   256,   784,   784,   784,    64,    64},
+    {   64,    64, 12544, 12544, 12544,    64,    64},
+    {  832,   256,    49,    49,    49,   832,   832},
+    {    9,    16, 23040, 23040, 23040,     9,     9}
+};
+
+// vector of vector, each vector is a {M, N, K, lda, ldb, ldc, ldd};
+const vector<vector<int>> resnet50_fwd_size_range = {
+    {12544,  1024,   256, 12544,   256, 12544, 12544},
+    {12544,  1024,   512, 12544,   512, 12544, 12544},
+    {12544,   256,  1024, 12544,  1024, 12544, 12544},
+    {12544,   256,   512, 12544,   512, 12544, 12544},
+    {12544,    64,   147, 12544,   147, 12544, 12544},
+    {  196,   256,  2304,   196,  2304,   196,   196},
+    { 3025,    64,   576,  3025,   576,  3025,  3025},
+    { 3136,  2048,  1024,  3136,  1024,  3136,  3136},
+    { 3136,  2048,   512,  3136,   512,  3136,  3136},
+    { 3136,   512,  1024,  3136,  1024,  3136,  3136},
+    { 3136,   512,  2048,  3136,  2048,  3136,  3136},
+    { 3136,    64,   576,  3136,   576,  3136,  3136},
+    {   49,   512,  4608,    49,  4608,    49,    49},
+    {50176,   128,   256, 50176,   256, 50176, 50176},
+    {50176,   512,   256, 50176,   256, 50176, 50176},
+    {  784,   128,  1152,   784,  1152,   784,   784}
+};
+
+// vector of vector, each vector is a {M, N, K, lda, ldb, ldc, ldd};
+const vector<vector<int>> resnet50_bwdwrw_size_range = {
+    { 1024,  2048,    49,    49,    49,  1024,  1024},
```

Comment:
I forget, does this K need to be a multiple of 4? Is the above value of 49 skipped, or is K allowed to not be a multiple of 4?

Reply:
Currently the test will return invalid (which passes silently).  This was something I meant to ask about but forgot to.

Reply:
@yoichiyoshida  does the test check for error code for non-multiple 4? can you point me to line number

Reply:
are all the *int8* test cases passing on your side?

Reply:
The multiple of 4 check occurs here:

https://github.com/ROCmSoftwarePlatform/rocBLAS/blob/develop/library/src/blas_ex/rocblas_gemm_ex.cpp#L441

Reply:
OK, I assume it is returning    rocblas_status_not_implemented

Reply:
It returns rocblas_status_invalid_size.  However I think these silently pass testing.  Was that intentional or a bug?

Reply:
The call to invalid k sizes currently returns rocblas_status_invalid_size. There's a switch statement in the generic test class that checks various cases, but not this one (because it is int8 specific). However the test currently just passes for cases not checked, so right now it currently passes even though we don't check the return code for this case. For the refactor, we can either remove the invalid cases, or explicitly check for the return code.

Reply:
Thanks, for a refactor, it will be be best to leave the test case and explicitly check the return code.

Reply:
yes, infact there should be category of tests that main purpose checks for error code returning sizes and make sure they return the appropriate codes; all normal cases should fail in gtest run if returning error code